### PR TITLE
Allow null-Values for OUTPUT parameter

### PR DESCRIPTION
**Pull request by boindiil**
Currently it is not possible to run stored procedures with sets OUTPUT parameter to null. With this change it is possible if the given type allows null values.

### CODE REVIEW

File: StoredProcedureEFCore/OutputParameter.cs [hunk 1]
```diff
@@ -3,17 +3,31 @@
...
+        public OutputParam(DbParameter param)
+        {
+            _dbParam = param;
+        }
+
+        public T Value
+        {
+            get
+            {
+                if (_dbParam.Value is System.DBNull)
```
Comment (verdie-g) [inline StoredProcedureEFCore/OutputParameter.cs]: Remove the System. Also, are you sure that this condition works ?

File: StoredProcedureEFCore/OutputParameter.cs [hunk 2]
```diff
@@ -3,17 +3,31 @@
...
+                {
+                    if (Nullable.GetUnderlyingType(typeof(T)) != null)
```
Comment (verdie-g) [inline StoredProcedureEFCore/OutputParameter.cs]: You can put this check in the first if

File: StoredProcedureEFCore/OutputParameter.cs [hunk 3]
```diff
@@ -3,17 +3,31 @@
...
+                    {
+                        return default(T);
+                    }
+                }
 
-    public T Value => (T)Convert.ChangeType(_dbParam.Value, typeof(T));
+                return (T)Convert.ChangeType(_dbParam.Value, typeof(T));
```
Comment (verdie-g) [inline StoredProcedureEFCore/OutputParameter.cs]: I'll have to check why I've used a ChangeType here. A cast should probably be enough.

Comment (boindiil) [reply StoredProcedureEFCore/OutputParameter.cs]: I have tested it. A cast ist not sufficient, for example dbo.ReturnBoolean (Program.cs line 39) would throw an exception as it is not possible to cast the int (bit on sql server) value to a boolean value.

File: StoredProcedureEFCore/OutputParameter.cs (hunk 1 above)
Comment (boindiil) [reply StoredProcedureEFCore/OutputParameter.cs]: I have removed the system and updated the if clause.

File: StoredProcedureEFCore/OutputParameter.cs (hunk 2 above)
Comment (boindiil) [reply StoredProcedureEFCore/OutputParameter.cs]: The reference check is now implicit. I have added the exception. I have expanded the correxponding tests.

**Comment by boindiil** [comment]
I have pushed new changes. What do I have to do that the status changes from "Change requested"?

**Comment by verdie-g** [comment]
Sorry I'm very busy, I might take care of this next week-end

**Comment by verdie-g** [comment]
Merged. Thanks for taking the time to add the stored procedure.

**Pull request merged**